docs/theming.md

```diff
@@ -0,0 +1,299 @@
+in the ShadyCSS README for details.
+
+## Colors
+
+Material Design theming uses two main colors: **primary** and **secondary**. The primary color is used
+throughout most of the application and components, as the main color for your application. The secondary color is used
+for floating action buttons and other interactive elements, serving as visual contrast to the primary.
+
+In addition to the primary and secondary colors, Material Theming also defines a _background_ color for the app, 
+as well as a _surface_ color, which is used used as a background in components.
+
+Finally, Material Theming has a number of text colors, which are used for rendering text and other shapes on top of the primary,
+secondary and background colors. These are specified as either dark or light, in order to provide sufficient contrast to
+what's behind them, and have
+[different levels of opacity depending on usage](https://material.io/design/color/the-color-system.html#):
+- Primary, used for most text.
+- Secondary, used for text which is lower in the visual hierarchy.
+- Hint, used for text hints (such as those in text fields and labels).
+- Disabled, used for text in disabled components and content.
+- Icon, used for icons.
+- On-surface, used for text that is on top of a surface background.
+- On-secondary, used for text that is on top of a secondary background.
+- On-primary, used for text that is on top of a primary background.
+
+### Color properties
+
+These custom properties used by MWC set the main colors for an application:
+
+| Custom property               | Description                                 |
+| ----------------------------- | ------------------------------------------- |
+| `--mdc-theme-primary`         | The theme primary color.                    |
+| `--mdc-theme-secondary`       | The theme secondary color.                  |
+| `--mdc-theme-surface`         | The theme surface color.                    |
+| `--mdc-theme-background`      | The theme background color.                 |
+
+When you set the main colors, you'll also have to set contrasting colors for text and iconography:
+
+| Custom property                               | Description                                                                |
+| --------------------------------------------- | -------------------------------------------------------------------------- |
+| `--mdc-theme-on-primary`                      | Text and icons on top of a theme primary color background.                   |
+| `--mdc-theme-on-secondary`                    | Text and icons on top of a theme secondary color background.                 |
+| `--mdc-theme-on-surface`                      | Text and icons on top of a theme surface color background.                      |
+
+The same pattern is followed for text colors on _background_:
+
+| Custom property                            | Description                                                |
```

Comment:
@asyncLiz do these properties actually get used or defined anywhere? I tried them out in a project to no avail.
Then I searched in the MDC and MWC repos, and the only hit I found was in the MDC theming doc. 

I'm thinking these are maybe not defined as CSS properties at all, and perhaps these values only make sense in the Sass world where you have more sophisticated theming?

Reply:
No, they're never defined anywhere. The only time they would be present (and usable) is if a component defined a color as something like `@include color(text-primary-on-background);`. Then the custom property would be generated.

Specifically for background, I don't think any of our components use it by default.

This is why I really want to overhaul the custom properties for MDC's theme. They desperately need updated to M2 and they need clearer documentation of what they're for.

In Sass, these keys are intended to automatically pick either "on light" or "on dark" based on what the background is. When that translates to mwc though, "on background" is the exact same as "on light" because Sass isn't used at runtime to determine the hue of the background color.

We're also missing the "on surface" variants. Like I said, it needs cleaned up.

Reply:
OK. So, it seems like for MWC I should remove (or at least comment out) the tables for `--mdc-theme-text-*`, since they'll never be set or used, and are only likely to confuse the end user.

Reply:
Hm. I know there are some of them that are used in components. I did a quick search and formfield uses `--mdc-theme-text-primary-on-background` for its label color. List also uses primary, secondary, hint, and icon on-background properties.

Maybe comment out until we clean up the theme properties?

Reply:
@asyncLiz Can you make an issue for this?

Reply:
OK, for now, I'll remove the `-on-dark` and `on-light` from this doc but leave the `on-background` colors which are used directly in some components.

Reply:
https://github.com/material-components/material-components-web/issues/5784 for the upstream issue. @dfreedm do you want another issue in MWC to track, or should it be linked to one of the existing theming issues?

Reply:
Commented out for now.